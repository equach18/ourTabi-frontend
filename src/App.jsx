import React from "react";
import "./App.css";
import { BrowserRouter } from "react-router-dom";
import RouteList from "./routes-nav/RouteList";
import NavBar from "./routes-nav/NavBar";
import UserProvider from "./context/UserProvider";

function App() {
  return (
    <BrowserRouter>
      <UserProvider>
          <div className="App">
            <NavBar />
            <RouteList />
          </div>
      </UserProvider>
    </BrowserRouter>
  );
}

export default App;
