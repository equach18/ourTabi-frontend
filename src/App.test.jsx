import { render } from "@testing-library/react";
import App from "./App";

it("renders correctly and matches snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
