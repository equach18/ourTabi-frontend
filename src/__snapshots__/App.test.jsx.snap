// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-gradient-to-r from-stone-300 via-white to-stone-300"
  >
    <nav
      class="bg-stone-300 shadow px-6 py-4 flex items-center justify-between relative"
    >
      <a
        aria-current="page"
        class="text-2xl font-bold text-orange-600 font-heading z-10 hover:text-orange-500 transition active"
        data-discover="true"
        href="/"
      >
        OurTabi
      </a>
      <div
        class="flex items-center space-x-4 z-10"
      >
        <a
          class="text-slate-600 hover:text-orange-500 transition"
          data-discover="true"
          href="/login"
        >
          Login
        </a>
        <a
          class="text-slate-600 hover:text-orange-500 transition"
          data-discover="true"
          href="/signup"
        >
          Sign Up
        </a>
      </div>
    </nav>
    <div
      class="min-h-[calc(100vh-4rem)] flex flex-col items-center justify-center px-6 py-12"
    >
      <div
        class="flex flex-col md:flex-row items-center justify-center gap-10 max-w-6xl w-full"
      >
        <div
          class="w-full max-w-xl text-center md:text-left space-y-6"
        >
          <p
            class="text-sm uppercase tracking-wide text-emerald-600 font-semibold"
          >
            Your shared journey begins here
          </p>
          <h1
            class="text-4xl font-bold text-orange-500"
          >
            Welcome to OurTabi!
          </h1>
          <p
            class="text-lg text-zinc-800"
          >
            Plan your trips, invite friends, and create unforgettable travel experiences. Make travel planning effortless at OurTabi.
          </p>
          <div
            class="space-x-4"
          >
            <a
              class="bg-slate-500 text-white px-6 py-3 rounded-full shadow-md hover:bg-slate-600 transition-transform hover:scale-105"
              data-discover="true"
              href="/login"
            >
              Login
            </a>
            <a
              class="bg-emerald-600 text-white px-6 py-3 rounded-full shadow-md hover:bg-emerald-700 transition-transform hover:scale-105"
              data-discover="true"
              href="/signup"
            >
              Sign Up
            </a>
          </div>
        </div>
        <div
          class="relative group w-full md:w-1/2 h-[580px] rounded-xl overflow-hidden shadow-xl border border-stone-200"
        >
          <img
            alt="Japanese temple"
            class="w-full h-full object-cover transition duration-500 group-hover:opacity-0"
            src="/src/assets/home.jpg"
          />
          <div
            class="absolute inset-0 bg-white text-zinc-700 p-6 flex flex-col justify-center items-center opacity-0 transition duration-800 group-hover:opacity-400"
          >
            <h3
              class="text-xl font-semibold text-orange-500 mb-4"
            >
              Why OurTabi?
            </h3>
            <ul
              class="space-y-3 text-sm text-center"
            >
              <li>
                Plan group trips easily
              </li>
              <li>
                Vote on activities
              </li>
              <li>
                Chat with trip members
              </li>
              <li>
                Explore other itineraries
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
